website/server/models/challenge.js

```diff
@@ -340,7 +340,7 @@ schema.methods.closeChal = async function closeChal (broken = {}) {
     // Set the challenge tag to non-challenge status and remove the challenge from the user's challenges
     User.update({
       challenges: challenge._id,
-      'tags._id': challenge._id,
+      'tags.id': challenge._id,
```

Comment:
This update query will fail to remove the challenge from the user's challenges if the user has deleted the challenge tag. That's something that users are able to do in case they prefer to use their own tag system and don't want the challenge tags cluttering their tag list.

Reply:
Can we just remove the tags.id part of the query here?

Reply:
We'd then be executing this for the case where the challenge tag had already been deleted:
`$set: {'tags.$.challenge': false},`
I.e., the challenge tag would be recreated, which would be a bug from the user's perspective. I'm on the fence about which is the worse bug. They're both pretty minor in my opinion. 

I could put up with the at worst very rare bug reports we'd get if the challenge tag was recreated. We might well never hear about it from any users. I'd like a comment in the code though just documenting that this situation could happen. I think that would be an easier "fix" than having two updates.

Reply:
In that case we may simply split this update into two: one that remove the challenge from user.challenges, the other that removes the tag where it exists

Reply:
If the user updates the tag then it would not be a challenge tag right? Then we don't have the issue this PR fixes

Reply:
Can you explain? What do you mean by `if the user updates`?

Reply:
Oh nvm. I see that the issue is the users challenge array, not the tags.